fe/fe-core/src/main/java/org/apache/doris/http/meta/MetaService.java

```diff
@@ -173,7 +173,27 @@ public void executeGet(BaseRequest request, BaseResponse response) {
                 writeResponse(request, response, HttpResponseStatus.BAD_REQUEST);
                 return;
             }
-            checkLongParam(versionStr);
+
+            long version = checkLongParam(versionStr);
+
+            // for master node, reject image put
+            if (Catalog.getCurrentCatalog().isMaster()) {
+                response.appendContent("this node is master, reject image put");
+                writeResponse(request, response, HttpResponseStatus.BAD_REQUEST);
+                LOG.error("this node is master, but receive image put from host{}, reject it", machine);
+                return;
+            }
+
+            // do not accept image whose version is bigger than max journalId
+            // if accepted, newly added log will not be replayed when restart
+            long maxJournalId = Catalog.getCurrentCatalog().getMaxJournalId();
+            if (version > maxJournalId) {
+                response.appendContent("image version is bigger than local max journal id, reject image put");
```

Comment:
better to add the value of max journal id and image id in response msg for easy debugging.